### PR TITLE
Add Ways To Provide SPIR-V Chapter

### PR DESCRIPTION
**Pull request by spencer-lunarg**
For tools, it is not obvious the various ways you can get SPIR-V into the driver, this adds a chapter to list them and give simple examples to help aid other tools (and maybe some apps)

### CODE REVIEW

File: chapters/ways_to_provide_spirv.adoc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+== vkCreateRayTracingPipelinesKHR::deferredOperation
+
+When dealing with Ray Tracing pipelines, it is important to note that a `VkDeferredOperationKHR` handle might be used to defer the creation of the pipeline to unblock the CPU thread.
+
+The link:https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#deferred-host-operations-requesting[spec states]
+
+> Parameters to the command requesting a deferred operation may be accessed by the implementation at any time until the deferred operation enters the complete state.
+
+This means if your tool is touching the SPIR-V being passed in, it must not take in consideration that the call to `vkCreateRayTracingPipelinesKHR` be asynchronous.
```
Comment (arno-lunarg) [inline chapters/ways_to_provide_spirv.adoc:149]:
```suggestion
In this particular case this means that if your tool is touching the SPIR-V being passed in, **all** parameters passed down to `vkCreateRayTracingPipelinesKHR`, including pointers, shader modules, inlined SPIR-V... must live until operation completion.
```

File: chapters/ways_to_provide_spirv.adoc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+The most common issue for tools or drivers consuming there is still information missing at `vkCreateShaderModule` time
+
+* For graphics pipeline, what are the descriptor set layouts bound
+* What are the other shader stages that will be linked up to this `VkShaderModule`
+* If there are multiple entry-points, which one is set (via `VkPipelineShaderStageCreateInfo::pName`)
+* What are the final Specialization Constant values
+
+== Inlining inside VkPipelineShaderStageCreateInfo
+
+If you have support for either `VK_KHR_maintenance5` or `VK_EXT_graphics_pipeline_library` are can just skip the `VkShaderModule` object all together.
```
Comment (spencer-lunarg) [inline chapters/ways_to_provide_spirv.adoc:41]:
```suggestion
If you have support for either `VK_KHR_maintenance5` or `VK_EXT_graphics_pipeline_library` you can just skip the `VkShaderModule` object completely.
```

File: chapters/ways_to_provide_spirv.adoc [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+VkShaderModule shader_module;
+vkCreateShaderModule(device, &shader_module_ci, NULL, &shader_module);
+
+// used to create VkPipeline
+VkPipelineShaderStageCreateInfo pipeline_stage_ci;
+pipeline_stage_ci.module = shader_module;
+----
+
+The most common issue for tools or drivers consuming there is still information missing at `vkCreateShaderModule` time
```
Comment (SaschaWillems) [inline chapters/ways_to_provide_spirv.adoc:32]: Not a native speaker, but this reads odd. Should there be a ":" before the "there"?

File: chapters/ways_to_provide_spirv.adoc [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+VkPipelineShaderStageCreateInfo pipeline_stage_ci;
+pipeline_stage_ci.pNext = &shader_module_id_ci
+pipeline_stage_ci.module = VK_NULL_HANDLE;
+----
+
+== Graphics Pipeline Library
+
+The `VK_EXT_graphics_pipeline_library` extension breaks up the pipeline into 4 smaller parts with the intent of allowing faster pipeline loading for applications reusing the same shaders or state in multiple pipelines.
+
+From a tools point a view, you may see up too 5 different `vkCreateGraphicsPipelines` calls, but only 2 of them will have SPIR-V in it. The following is an example workflow:
```
Comment (SaschaWillems) [inline chapters/ways_to_provide_spirv.adoc:97]: "to" instead of "too"

File: chapters/ways_to_provide_spirv.adoc [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+VkPipeline executable_pipeline;
+VkPipelineLibraryCreateInfoKHR library_ci;
+library_ci.pLibraries = [vertex_input_lib, pre_raster_lib, fragment_shader_lib, fragment_output_lib];
+create_info.pNext = &library_ci;
+vkCreateGraphicsPipelines(device, VK_NULL_HANDLE, 1, create_info, NULL, executable_pipeline);
+----
+
+== Shader Objects
+
+The `VK_EXT_shader_object` extension created a new complete flow to pass state information in the command buffer that doesn't involve pipelines. The following is the workflow to pass in the SPIR-V.
```
Comment (SaschaWillems) [inline chapters/ways_to_provide_spirv.adoc:125]: "Complete(ly) new flow" instead of "new complete flow"?

**Review by SaschaWillems — CHANGES_REQUESTED** [review]
Looking good. Only had some minor remarks on spelling :)

File: chapters/ways_to_provide_spirv.adoc (hunk 3 above)
Comment (spencer-lunarg) [reply chapters/ways_to_provide_spirv.adoc]: rephrased this, made new commit to see the changes, but thanks for helping me with my native language :laughing:

**Review by SaschaWillems — APPROVED** [review]

**Pull request merged**